dotCMS/src/integration-test/java/com/dotmarketing/db/DockerSecretDataSourceStrategyTest.java

```diff
@@ -0,0 +1,138 @@
+     * @throws SQLException
+     * @throws IOException
+     * @throws NamingException
+     */
+    @Test
+    public void testApply() throws SQLException, IOException, NamingException {
+        final HikariDataSource testDatasource = (HikariDataSource) TestInitialContext.getInstance()
+                .getDataSource();
+
+        final File tempFile = createTempFile(
+                "connection_db_driver=" + testDatasource.getDriverClassName() + "\n"
+                        + "connection_db_base_url=" + testDatasource.getJdbcUrl() + "\n"
+                        + "connection_db_username=" + testDatasource.getUsername() + "\n"
+                        + "connection_db_password=" + testDatasource.getPassword() + "\n"
+                        + "connection_db_validation_query=SELECT 1");
+
+        final SystemEnvironmentProperties systemEnvironmentProperties = Mockito.mock(SystemEnvironmentProperties.class);
+
+        Mockito.when(systemEnvironmentProperties.getVariable("DOCKER_SECRET_FILE_PATH")).thenReturn(tempFile.getPath());
+
+        final DockerSecretDataSourceStrategy strategy = new DockerSecretDataSourceStrategy(systemEnvironmentProperties);
+
+        final DataSource dataSource = strategy.apply();
+
+        assertNotNull(dataSource);
+        assertTrue(dataSource instanceof HikariDataSource);
+        assertNotNull(dataSource.getConnection());
+    }
+
+    /**
+     * Method to test: {@link DockerSecretDataSourceStrategy#getHikariConfig(Map)}
+     * Test case: Tests that all values in the config object are set correctly
+     * Expected result: Success
+     */
+    @Test
+    public void testGetHikariConfig(){
+        final Map<String, String> dockerSecretsMap = new HashMap<>();
+
+        dockerSecretsMap.put("connection_db_driver", "org.postgresql.Driver");
+        dockerSecretsMap.put("connection_db_base_url", "jdbc:postgresql://localhost/dotcms");
+        dockerSecretsMap.put("connection_db_username", "username");
+        dockerSecretsMap.put("connection_db_password", "password");
+        dockerSecretsMap.put("connection_db_max_total", "60");
+        dockerSecretsMap.put("connection_db_max_idle", "10");
+        dockerSecretsMap.put("connection_db_max_wait", "60000");
+        dockerSecretsMap.put("connection_db_validation_query", "SELECT 1");
+        dockerSecretsMap.put("connection_db_leak_detection_threshold", "60000");
+
+        final HikariConfig config = DockerSecretDataSourceStrategy.getInstance()
+                .getHikariConfig(dockerSecretsMap);
+
+        validateConfiguration(config, dockerSecretsMap);
+    }
+
+    /**
+     * Verifies all values in the config object are set correctly
+     * @param config {@link HikariConfig} to be validated
+     * @param dockerSecretsMap {@link Map} contains the expected values to be compared
+     */
+    private void validateConfiguration(final HikariConfig config, Map<String, String> dockerSecretsMap) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'dockerSecretsMap' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5026767)